simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/StationSimulatorFunctionalTest.java

```diff
@@ -637,6 +636,43 @@ void shouldGetHeartbeatIntervalWithGetVariablesRequest() {
         });
     }
 
+    @Test
+    void shouldSendNotifyReportOnGetBaseReportRequest() {
+        mockSuccessfulGetBaseReportAnswer();
+
+        stationSimulatorRunner.run();
+
+        GetBaseReportRequest request = new GetBaseReportRequest().withRequestId(200).withReportBase(FULL_INVENTORY);
+
+        String payload = new Call(UUID.randomUUID().toString(), ActionType.GET_BASE_REPORT, request).toJson();
+
+        await().untilAsserted(() -> {
+            List<Call> stationCalls = server.getReceivedCalls(STATION_ID);
+            assertThat(stationCalls.stream()).anyMatch(call -> call.getActionType() == ActionType.BOOT_NOTIFICATION);
+        });
+
+        stationSimulatorRunner.getStation(STATION_ID).sendMessage(new StationMessage(STATION_ID, StationMessage.Type.OCPP_MESSAGE, payload));
+
+        await().untilAsserted(() -> {
+            List<Call> stationCalls = server.getReceivedCalls(STATION_ID);
+
+            List<NotifyReportRequest> notifyReportRequests =
+                    stationCalls
+                            .stream()
+                            .filter(call -> call.getActionType() == ActionType.NOTIFY_REPORT)
+                            .map(call -> NotifyReportRequest.class.cast(call.getPayload()))
+                            .collect(toList());
+
+            sort(notifyReportRequests, new NotifyReportComparator());
+
+            int size = notifyReportRequests.size();
+            for (int i = 0; i < size; i++) {
+                assertThat(notifyReportRequests.get(i).getSeqNo()).isEqualTo(i);
+                assertThat(notifyReportRequests.get(i).getTbc()).isEqualTo(i != size - 1);
```

Comment:
Tests have to be readable. Please change that to:

`assertThat(notifyReportRequests.get(i).getTbc()).isFalse();`

and loop until size - 1

then do 

`assertThat(notifyReportRequests.get(size - 1).getTbc()).isTrue();`